docs/en/dgs/dataloader.md

```diff
@@ -1,3 +1,158 @@
-# Load data from source
+# Loading From Your Data Source
 
-(comming soon...)
+By using kafka queues as the entry for streaming graph updates, the Dynamic-Graph-Service can decouple the different
+data sources and sampling workers.
+you should develop your own dataloader to read you data source and produce the graph updates into output kafka queues
+for further consuming.
+
+## Rules to Follow
+
+When you process you source data and produce them in your dataloader, you should follow these rules:
+- The graph schema from your source data must be consistent with dgs.
+- A produced kafka message must be a batch of graph updates, defined as a flatbuffers table [RecordBatchRep](https://github.com/alibaba/graph-learn/blob/master/dynamic_graph_service/fbs/record.fbs).
+- Graph updates and partitioned for sampling workers, all records in one batch must have the same data partition id.
+
+## Dataloader SDK
+
+In practice, it is difficult for users to batch records with dgs graph schema, manage partitioning logic and
+produce partitioned batches into kafka queues with themselves.
+Thus, we provide a dataloader sdk (a c++ lib) to help users to do these things.
+
+
+### Build
+
+Build dataloader sdk with cmake:
+```shell
+$ cd dynamic_graph_service/dataloader
+$ mkdir build && cd build
+$ cmake -DCMAKE_INSTALL_PREFIX=$your_install_prefix ..
+$ make && make install
+```
+
+use it in your CMakeLists.txt:
+```cmake
+list (APPEND CMAKE_PREFIX_PATH $your_install_prefix)
+find_package (DataLoader)
+target_link_libraries(your_project_name
+  PUBLIC DataLoader::dataloader)
+```
+
+### Initialization
+You must initialize the dataloader sdk components before your program runs.
```

Comment:
The dataloader must be initialized before the program runs

Reply:
done